### DISCUSSION TITLE
Making the Pitch for Oqtane?

### DISCUSSION THREAD

**Discussion by Senyors** [oqtane/oqtane.framework#815] [Q&A]
I'm a .NET developer for a major state agency here in Wisconsin, where I've spent the past few years creating a custom .NET MVC 5 framework to drive several finance-based systems. However, with the advent of .NET Core, Blazor and the imminent release of .NET 5, it seems like now might be the time to pitch Oqtane to my management and leverage what a community of open source developers can do versus what I've developed solely on my own.

Back in the day, I actually pitched DNN to my management, but by then web forms was on its way out and I was asked to develop my own custom framework from scratch in MVC 5 instead. Now, I'm thinking about pitching Oqtane, but I'll need to convince some wary IT managers that something like Oqtane is not only ready for prime-time but will remain supported and available for the foreseeable future.

Does the community have any thoughts regarding how I should proceed? Put another way, is Oqtane truly ready to be leveraged by a major state organization (especially one that manages billions of dollars on an annual basis)? Or, should I perhaps wait on Oqtane and allow it a little more time to mature? And if I could convince my management to accept Oqtane, will there be significant training resources and support made available sometime soon? I've gone through those modules that Shaun created back in the summer, but my development team is brand new to .NET Core--are their recommendations as to how best to go up the appropriate learning curves in order to understand and work with Oqtane?

I would appreciate any and all feedback from the community!

**Comment by mikecasas** [comment]
Oqtane is ready in that it's built on .NET Core and Blazor, both of which seem to be top priorities for Microsoft.

What are you looking to build? One major app? Lots of smaller apps? 

It seems the bigger learning curve will be .NET Core for your team.

**Reply by Senyors** [reply]
When I first joined my team five years ago, I tried to pitch DNN in building a brand new, fairly grandiose major application that would consist of many modules. The pitch was ultimately rejected, however, because of DNN's reliance on web forms (our agency had since migrated to MVC). In the end, I created my own MVC-based framework and it's running several important modules now, just as we had planned. However, I struggled with MVC and, as you know, developing and maintaining a custom framework from scratch is incredibly difficult. 

Over the past few months, my agency has adopted .NET Core as a new standard (with several teams already creating Blazor-based applications), and with .NET 5 right around the corner, it really seems like a great time to pitch Oqtane. Ideally, I'd like to learn Oqtane well enough to champion it, then begin rebuilding our systems in Oqtane, creating a single large application to replace what I developed a few years ago (and then adding to it as needed over the foreseeable future). Does that sound like an appropriate business case for leveraging Oqtane?

**Reply by mikecasas** [reply]
Is your MVC framework in .NET or .NET Core? If you want to upgrade from .NET to Core and use oqtane then I would think that's a great business case.

**Reply by Senyors** [reply]
Yes, our existing framework is in .NET MVC 5, and my management is already interested in migrating to .NET Core. So yes, I'm also thinking that the business case is strong--I just need to learn Oqtane well enough to champion it within my organization I think.

**Comment by sbwalker** [comment]
@Senyors I am obviously biased but personally I feel that there is no purpose creating an open source framework unless people  feel confident using it in production applications. And the fact that Oqtane has been under development for 2 years and has already had a number of official 1.x releases should also make you feel more comfortable ( ie. it is not in a perpetual beta state ). Obviously, Oqtane is closely related to Blazor so I am not sure if part of your question relates to convincing management that Blazor is ready for primetime as well?

In regards to your other question... "will there be significant training resources and support made available sometime soon?" This is a great question... but like any other emerging technology, you are highlighting a larger dilemma in that there needs to be enough demand for people to invest in creating these types of resources, but without these types of resources available its also hard to create the demand. One thing I will point out, as I am not sure if it is 100% apparent to everyone, is that there are MANY conceptual similarities between DNN and Oqtane - which means that there are a ton of resources which have been created for DNN over the past 20 years which could be easily updated or adapted so that they are relevant to Oqtane.

**Reply by Senyors** [reply]
@sbwalker The good news on our end is that several other teams within my agency have already started building smaller applications in Blazor, so that precedent is already set. But our organization has a strange distrust of "outside" software (even though we leverage client-side tools like Telerik/Kendo and run our public-facing site on Drupal), so my business case to leverage Oqtane has to be incredibly strong. We also manage roles and security in-house, so that's a hoop I'd need to jump through in trying to bring Oqtane in-house too. 

For what it's worth, I'm coming at this from a strong DNN perspective. Back before .NET, I wrote several web content management systems in Classic ASP (one of which is still online today), then migrated all of that to DNN and, in its heyday, I was hosting dozens and dozens of DNN sites across multiple instances. I even got into custom application framework development with DNN and built several sites, one of which I'm still tinkering with today (https://browserquests.com). So, to me, this all makes perfect sense, and with my own history of DNN, my hope is to leverage all that expertise on behalf of what I'm now doing for my state agency.

That said, Shaun, I did give Oqtane a serious look back in the spring and I struggled with it; at heart, I'm a web forms developer and I'm still sort of stuck in that era. I think what I need are some resources to help me quickly climb the .NET Core / .NET 5 learning curve, so I can take another serious stab at Oqtane, really understand how it works and begin developing in it so I can, in turn, best pitch my own state agency with direct, hands-on expertise. Do you have any suggestions on how I should proceed?

**Reply by sbwalker** [reply]
@Senyors there is definitely a learning curve with the newer technology. More moving parts to consider when creating a client/server application. Since the spring  there are now additional assets to help people with getting started.... there are more examples, Module Creator, video series, API documentation, presentations, etc.... ( most of which are linked from the home page of this repo ) That being said, if Blazor Server meets your hosting requirements, it is also possible to build server-side modules in Oqtane with the same simplicity as WebForms. Perhaps we need a server-side example of the Blog module to demonstrate this.
